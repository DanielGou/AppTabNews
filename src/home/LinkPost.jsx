import { StyleSheet, Text, View } from 'react-native';

export default function LinkPost({ post }){
    function publishedAt(){
        let published_At = new Date(post.item.published_at);
        let actualDate = new Date();

        const utcPublished_At = Date.UTC(published_At.getFullYear(), published_At.getMonth(), published_At.getDate(), published_At.getHours());
        const utcActualDate = Date.UTC(actualDate.getFullYear(), actualDate.getMonth(), actualDate.getDate(), actualDate.getHours());
        
        let result = ((utcActualDate - utcPublished_At) / (1000 * 60 * 60));

        if(result > 24){
            let dias = `${Math.floor(result / 24)} `

            return dias + (result > 48 ? 'dias atrás': 'dia atrás')
        }

        return result + ' horas atrás'
    }

    return(
        <View style={styles.wrapperLinkPost}>
            <Text style={styles.title}>{post.index + 1}.</Text>
            <View style={styles.ml5}>
                <Text style={styles.title}>{post.item.title}</Text>
                <View style={styles.wrapperInfoPost}>
                    <Text>{post.item.children_deep_count} comentários  ·</Text>
                    <Text style={styles.ml5}>{post.item.username}  ·</Text>
                    <Text style={styles.ml5}>{publishedAt()}</Text>
                </View>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    wrapperLinkPost:{
        display: 'flex',
        flexDirection: 'row',
        margin: 15,
    },
    title:{
        fontSize: 17,
        fontWeight: 'bold'
    },
    ml5:{
        marginLeft: 5
    },
    wrapperInfoPost:{
        display: 'flex',
        flexDirection: 'row',
    }
})