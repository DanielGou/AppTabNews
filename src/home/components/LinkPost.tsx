import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default function LinkPost({ post }: { post: any; }): JSX.Element{
    const { title, username, children_deep_count, publishedAt } = post.item
 
    return(
        <View style={styles.wrapperLinkPost}>
            <Text style={styles.title}>{post.index + 1}.</Text>
            <View style={styles.ml5}>
                <Text style={styles.title}>{title}</Text>
                <View style={styles.wrapperInfoPost}>
                    <Text>{children_deep_count} comentários  ·</Text>
                    <Text style={styles.ml5}>{username}  ·</Text>
                    <Text style={styles.ml5}>{publishedAt}</Text>
                </View>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    wrapperLinkPost:{
        display: 'flex',
        flexDirection: 'row',
        margin: 15,
    },
    title:{
        fontSize: 17,
        fontWeight: 'bold'
    },
    ml5:{
        marginLeft: 5
    },
    wrapperInfoPost:{
        display: 'flex',
        flexDirection: 'row',
    }
})