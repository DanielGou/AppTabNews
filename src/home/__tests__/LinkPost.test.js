import React from "react";
import { create } from 'react-test-renderer';

import LinkPost from "../components/LinkPost";

const post ={
    index: 0,
    item:{
        title: 'Titulo', 
        username: 'Daniel', 
        children_deep_count: '0 comentarios', 
        publishedAt: '1 hora atr√°s'        
    }
}

const tree = create(<LinkPost post={post} />)

test('snapshot Home', ()=>{
    expect(tree).toMatchSnapshot()
})
