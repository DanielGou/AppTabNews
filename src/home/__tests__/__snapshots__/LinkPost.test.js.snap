// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Home 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "margin": 15,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 17,
        "fontWeight": "bold",
      }
    }
  >
    1
    .
  </Text>
  <View
    style={
      Object {
        "marginLeft": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 17,
          "fontWeight": "bold",
        }
      }
    >
      Titulo
    </Text>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text>
        0 comentarios
         comentários  ·
      </Text>
      <Text
        style={
          Object {
            "marginLeft": 5,
          }
        }
      >
        Daniel
          ·
      </Text>
      <Text
        style={
          Object {
            "marginLeft": 5,
          }
        }
      >
        1 hora atrás
      </Text>
    </View>
  </View>
</View>
`;
